tests/cupy_tests/sorting_tests/test_search.py

```diff
@@ -431,3 +431,73 @@ def test_nanargmax_zero_size_axis0(self, xp, dtype):
     def test_nanargmax_zero_size_axis1(self, xp, dtype):
         a = testing.shaped_random((0, 1), xp, dtype)
         return xp.nanargmax(a, axis=1)
+
+
+@testing.gpu
+@testing.parameterize(*testing.product(
+    {'bins': [
+        [],
+        [0, 1, 2, 4, 10],
+        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        [0.0, 1.0, 2.5, 4.0, 10.0],
+        [-1.0, 1.0, 2.5, 4.0, 20.0],
+        [1.5, 2.5, 4.0, 6.0],
+        [0.0, 1.0, 1.0, 4.0, 4.0, 10.0],
+        [0.0, 1.0, 1.0, 4.0, 4.0, 4.0, 4.0, 10.0],
+    ],
+        'side': ['left', 'right'],
+        'shape': [(), (10,), (6, 3, 3)]})
+)
+class TestSearchSorted(unittest.TestCase):
+
+    @testing.with_requires('numpy>=1.10.0')
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_list_equal()
+    def test_searchsorted(self, xp, dtype):
+        x = testing.shaped_arange(self.shape, xp, dtype)
+        bins = xp.array(self.bins)
+        y = xp.searchsorted(bins, x, side=self.side)
+        return y,
+
+
+@testing.gpu
+class TestSearchSortedInvalid(unittest.TestCase):
+
+    # Cant test unordered bins due to numpy undefined
+    # behavior for searchsorted
+    @testing.with_requires('numpy>=1.10.0')
+    @testing.numpy_cupy_raises(accept_error=ValueError)
+    def test_searchsorted_ndbins(self, xp):
+        x = testing.shaped_arange((10,), xp, xp.float64)
+        bins = xp.array([[10, 4], [2, 1], [7, 8]])
+        y = xp.searchsorted(bins, x)
+        return y,
+
+
+@testing.gpu
+class TestSearchSortedWithSorter(unittest.TestCase):
+
+    @testing.with_requires('numpy>=1.10.0')
+    @testing.numpy_cupy_array_list_equal()
+    def test_sorter(self, xp):
+        x = testing.shaped_arange((10,), xp, xp.float64)
```

Comment:
The maximum value in the array `x` should be greater than 10 for coverage.
```suggestion
        x = testing.shaped_arange((12,), xp, xp.float64)
```